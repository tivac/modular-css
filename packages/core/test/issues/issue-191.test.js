"use strict";

var fs = require("fs"),
    
    namer = require("test-utils/namer.js"),

    Processor = require("../../processor.js");

describe("/issues", function() {
    describe("/191", function() {
        var fn = it;
        
        afterAll(() => require("shelljs").rm("-rf", "./packages/core/test/output/sensitive.txt"));

        // Verify that filesystem is case-insensitive before bothering
        fs.writeFileSync("./packages/core/test/output/sensitive.txt");

        try {
            fs.statSync("./packages/core/test/output/SENSITIVE.txt");
        } catch(e) {
            fn = it.skip;
        }

        fn("should ignore case differences in file paths", function() {
            var processor = new Processor({
                    namer
                });
            
            return processor.file(require.resolve("./specimens/191/start.css"))
                .then(() => processor.output())
                .then((result) => expect(result.css).toMatchSnapshot());
        });
    });
});
