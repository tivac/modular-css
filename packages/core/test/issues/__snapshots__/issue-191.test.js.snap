// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/issues /191 should ignore case differences in file paths 1`] = `
"/* packages/core/test/issues/specimens/191/values.css */
.fooga {
    color: #F00;
}

/* packages/core/test/issues/specimens/191/start.css */
"
`;
